### DISCUSSION TITLE
check that the public keys lie on the curve.?

### DISCUSSION THREAD

**Discussion by Noname400** [tlsfuzzer/python-ecdsa#274] [Q&A] [answered]
hello, tell me how to check that the public keys lie on the curve.?
and how to check that after the operation of multiplication I left the border of the curve?

**Comment by tomato42** [comment, marked as answer]
All the public methods in VerifyingKey check if the public key lies on the curve, it's controlled by the `validate_point` option like:
https://github.com/tlsfuzzer/python-ecdsa/blob/b3b27cd4811ce935e42bbcd251206ff2cb3b51b9/src/ecdsa/keys.py#L202-L204
or:
https://github.com/tlsfuzzer/python-ecdsa/blob/b3b27cd4811ce935e42bbcd251206ff2cb3b51b9/src/ecdsa/keys.py#L286-L293

> and how to check that after the operation of multiplication I left the border of the curve?

if a point was on the curve before multiplication, then the result will be on the curve...

You can double check that on high level, by exporting the point to a byte string and then create a new VerifyingKey instance, that will verify if it is on curve still.

On low level, you need to get the `CurveFp()` and call `contains_point()` for short Weierstrass curves:
https://github.com/tlsfuzzer/python-ecdsa/blob/b3b27cd4811ce935e42bbcd251206ff2cb3b51b9/src/ecdsa/ellipticcurve.py#L131
or the `contains_point()` of `CurveEdTw()` for twisted Edwards curves:
https://github.com/tlsfuzzer/python-ecdsa/blob/b3b27cd4811ce935e42bbcd251206ff2cb3b51b9/src/ecdsa/ellipticcurve.py#L197
